Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm

```diff
@@ -262,24 +262,36 @@ static String defaultWebsiteDataStoreRootDirectory()
 
 void WebsiteDataStore::fetchAllDataStoreIdentifiers(CompletionHandler<void(Vector<UUID>&&)>&& completionHandler)
 {
-    Vector<UUID> identifiers;
-    for (auto identifierString : FileSystem::listDirectory(defaultWebsiteDataStoreRootDirectory())) {
-        if (auto identifier = UUID::parse(identifierString))
-            identifiers.append(*identifier);
-    }
+    ASSERT(isMainRunLoop());
 
-    completionHandler(WTFMove(identifiers));
+    websiteDataStoreIOQueue().dispatch([completionHandler = WTFMove(completionHandler), directory = defaultWebsiteDataStoreRootDirectory().isolatedCopy()]() mutable {
+        Vector<UUID> identifiers;
+        for (auto& identifierString : FileSystem::listDirectory(directory)) {
```

Comment:
Would be nice to use compactMap() for this. It would be a bit shorter and more efficient.

Likely something like:
```
auto identifiers = WTF::compactMap([](auto& identifierString) { return UUID::parse(identifierString); });
```